src/modules/category/category.service.ts

```diff
@@ -0,0 +1,40 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { CategoryDto } from '../database/dto/category.dto';
+import { Category } from '../database/entities/category.entity';
+
+@Injectable()
+export class CategoryService {
+  constructor(
+    @InjectRepository(Category)
+    private readonly categoryRepository: Repository<Category>,
+  ) {}
+
+  async getCategories(): Promise<CategoryDto[]> {
+    return this.categoryRepository.find();
+  }
+
+  async getCategoryById(id: string): Promise<CategoryDto> {
+    return this.categoryRepository.findOne(id);
+  }
+
+  async createCategory(category: CategoryDto): Promise<CategoryDto> {
+    const newCategory = await this.categoryRepository.create(category);
+    await this.categoryRepository.save(newCategory);
+    return newCategory;
+  }
+
+  async updateCategory(id: string, category: CategoryDto): Promise<CategoryDto> {
+    await this.categoryRepository.update(id, category);
+    return this.categoryRepository.findOne(id);
```

Comment:
update with returning

Reply:
return this.categoryRepository.save({ id, category });